File: core/src/main/java/com/mcreater/amclcore/concurrent/task/model/EmptyAction.java
package com.mcreater.amclcore.concurrent.task.model;

import com.mcreater.amclcore.concurrent.task.AbstractAction;
import com.mcreater.amclcore.i18n.Text;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;

import static com.mcreater.amclcore.i18n.I18NManager.translatable;

@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class EmptyAction extends AbstractAction {
    protected void execute() {
    }

    protected Text getTaskName() {
        return translatable("core.concurrent.task.model.empty");
    }

    public static EmptyAction of() {
        return new EmptyAction();
    }
}

File: core/src/main/java/com/mcreater/amclcore/concurrent/task/model/EmptyAction.kt
package com.mcreater.amclcore.concurrent.task.model

import com.mcreater.amclcore.concurrent.task.AbstractAction
import com.mcreater.amclcore.i18n.I18NManager.Companion.translatable
import com.mcreater.amclcore.i18n.Text


class EmptyAction : AbstractAction() {
    override fun execute() {}
    override fun getTaskName(): Text {
        return translatable("core.concurrent.task.model.empty")
    }

    companion object {
        @JvmStatic
        fun of(): EmptyAction {
            return EmptyAction()
        }
    }
}

